File: Web/src/api/userGroup.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetUserGroupList(data) {
  return apiClient.post("/userGroup/GetUserGroupList", data).then(handleResponse).catch(handleError)
}

function GetVisibleUserGroupList(data) {
  return apiClient.post("/userGroup/GetVisibleUserGroupList", data).then(handleResponse).catch(handleError)
}

function AddUserGroup(data) {
  return apiClient.post("/userGroup/AddUserGroup", data).then(handleResponse).catch(handleError)
}

function RemoveUserGroup(data) {
  return apiClient.post("/userGroup/RemoveUserGroup", data).then(handleResponse).catch(handleError)
}

const userGroupApi = {
  GetUserGroupList,
  GetVisibleUserGroupList,
  AddUserGroup,
  RemoveUserGroup
}

export default userGroupApi

File: Server/route/userGroup.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserGroupRoute(r *gin.Engine) {
	userGroup := r.Group("/userGroup")
	userGroup.Use(middleware.JWTAuthMiddleware())
	{
		userGroup.POST("/GetUserGroupList", api.GetUserGroupList)
		userGroup.POST("/GetVisibleUserGroupList", api.GetVisibleUserGroupList)
		userGroup.POST("/AddUserGroup", api.AddUserGroup)
		userGroup.POST("/RemoveUserGroup", api.RemoveUserGroup)
	}
}
